lc-0125-valid-palindrome/src/test/java/solution/CustomCsvArgTest.java

```diff
@@ -0,0 +1,33 @@
+package solution;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CustomCsvArgTest {
+
+    static class MyBoolean {
+        public boolean boolVal;
+
+        MyBoolean(String s) {
+            if (s.equals("true")) {
+                boolVal = true;
+            } else if (s.equals("false")) {
+                boolVal = false;
+            }
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "A man, a plan, a canal: Panama; true",
+            "race a car; false",
+            "''; true"
```

Comment:
These are fun:
- "()()"
- ")((()"

try them :)

Reply:
Added to testcases and updated the code to handle these.